Slingshot.Api/Concrete/BitbucketRepository.cs

```diff
@@ -234,6 +242,25 @@ public BitbucketRepository(Uri uri, string host, string token)
             }
         }
 
+        public override async Task<BitbucketRepository> GetPullRequest<BitbucketRepository>(string prId)
+        {
+            try
+            {
+                SourceControlInfo info = await this.GetScmInfo();
+                string repoToken = info.token;
+
+                using (HttpClient client = CreateHttpClient(accessToken: repoToken))
```

Comment:
right. there is only one exception, when checking a repo is private or not we will call "repository" API without token.
I think the best is to create another helper function to include the logic to get access token. So that the private check don`t need to get token to save one call.